modular_skyrat/modules/customization/modules/clothing/storage/belts.dm

```diff
@@ -0,0 +1,115 @@
+/obj/item/storage/belt/crusader	//Belt + sheath combination - still only holds one sword at a time though
+	icon = 'modular_skyrat/modules/customization/icons/obj/clothing/storage.dmi'
+	worn_icon = 'modular_skyrat/modules/customization/icons/mob/clothing/storage.dmi'
+	name = "crusader belt"
+	desc = "Holds an assortment of equipment for whatever situation an adventurer may encounter, as well as having an attached sheath."
+	icon_state = "crusader_belt"
+	worn_icon_state = "crusader_belt"
+	inhand_icon_state = "utility"
+	w_class = WEIGHT_CLASS_BULKY //Cant fit a sheath in your bag
+	component_type = /datum/component/storage/concrete/belt/crusader
+
+//Credit to Funce for this chunk of code directly below, which overrides normal dumping code and instead dumps from the pouch item inside
+/datum/component/storage/concrete/belt/crusader/dump_content_at(atom/dest_object, mob/M)
+    var/atom/A = parent
+    var/atom/dump_destination = dest_object.get_dumping_location()
+    if(A.Adjacent(M) && dump_destination && M.Adjacent(dump_destination))
+        var/obj/item/storage/belt/storage_pouch/pouch = locate() in real_location()
+        if (!pouch)
+            to_chat(M, "<span class='warning'>[parent] doesn't seem to have a pouch to empty.</span>")
+            return FALSE //oopsie!! If we don't have a pouch! You're fucked!
+        var/datum/component/storage/STR = pouch.GetComponent(/datum/component/storage)
+        if(locked)
+            to_chat(M, "<span class='warning'>[parent] seems to be locked!</span>")
+            return FALSE
+        if(dump_destination.storage_contents_dump_act(STR, M))
+            playsound(A, "rustle", 50, TRUE, -5)
+            A.do_squish(0.8, 1.2)
+            return TRUE
+    return FALSE
+
+/obj/item/storage/belt/crusader/CtrlClick(mob/user)	//Makes ctrl-click also open the inventory, so that you can open it with full hands without dropping the sword
+	. = ..()
+	var/datum/component/storage/STR = GetComponent(/datum/component/storage)
+	STR.user_show_to_mob(user)
+	return
+
+/obj/item/storage/belt/crusader/AltClick(mob/user)	//This is basically the same as the normal sheath, but because there's always an item locked in the first slot it uses the second slot for swords
+	if(!user.canUseTopic(src, BE_CLOSE, NO_DEXTERITY, FALSE, TRUE))
+		return
+	if(contents.len == 2)
```

Comment:
I'm not a fan of this snowflake way of checking the contents ;-; see if you can redo it?

Reply:
Maybe have it as a variable on the sheath itself?

Reply:
honestly having so much trouble wrapping my head around changing this, i based it off the sheath which only had 1 slot and only checked if the contents were filled up to that one.,
I'll get to this later, but for now I'm pushing the changes that I DID understand lmao